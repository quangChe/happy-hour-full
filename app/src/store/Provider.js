import React from 'react';
import { Provider as ReduxProvider} from 'react-redux';
import Store from '../store';

const Provider = (props) => (
  <ReduxProvider store={Store}>{props.children}</ReduxProvider>
);

export default Provider;

