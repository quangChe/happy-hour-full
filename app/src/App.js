import React from 'react';
import { fetchNearby } from './store/actions';
import { connect } from 'react-redux';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import HomeScreen from './screens/HomeScreen/HomeScreen';
import NavBar from './components/NavBar/NavBar';
import ErrorScreen from './screens/ErrorScreen/ErrorScreen';


class App extends React.Component {
  componentDidMount() {
    this.props.fetchNearby();
  }

  render() {
    return (
      <Router>
        <NavBar/>
        <Switch>
          <Route path="/" exact component={HomeScreen}/>
          <Route component={ErrorScreen}/>
        </Switch>
      </Router>
    )
  }
}

export default connect(null, {fetchNearby})(App);
