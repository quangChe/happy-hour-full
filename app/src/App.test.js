import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import Provider from './store/Provider';

describe('<App />', () => {
  const app = shallow(<Provider><App /></Provider>);

  it('renders without crashing', () => {
    expect(app).toMatchSnapshot();
  });

  describe('on startup', () => {
    const mounted = mount(<Provider><App /></Provider>);
    it('renders Router', () => {
      expect(mounted.find('Router').exists()).toBe(true);
    });
  
    it('renders Navbar', () => {
      expect(mounted.find('NavBar').exists()).toBe(true);
    });
  
    it('renders the Home screen', () => {
      expect(mounted.find('HomeScreen').exists()).toBe(true);
    });

  });
  
});
