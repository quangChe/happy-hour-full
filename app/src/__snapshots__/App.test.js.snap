// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction />
</Provider>
`;
